pdg/src/builder.rs

```diff
@@ -49,23 +61,15 @@ impl EventKindExt for EventKind {
             Alloc { ptr, .. } => ptr,
             AddrOfLocal(lhs, _) => lhs,
             Offset(ptr, _, _) => ptr,
-            Done => return None,
+            Done | BeginBody => return None,
         })
     }
 
-    fn has_parent(&self) -> bool {
-        use EventKind::*;
-        !matches!(
-            self,
-            Realloc { new_ptr: _, .. } | Alloc { ptr: _, .. } | AddrOfLocal(_, _) | Done
-        )
-    }
-
-    fn parent(&self, obj: (GraphId, NodeId)) -> Option<(GraphId, NodeId)> {
-        self.has_parent().then_some(obj)
-    }
-
-    fn to_node_kind(&self) -> Option<NodeKind> {
+    fn to_node_kind(
+        &self,
+        func: Func,
+        address_taken: &mut IndexSet<(Func, u32)>,
```

Comment:
Can the `u32` just be `Local`?  If not, we should at least use a type alias for clarity.